#161255353 - User Signup/Registration validation errors

AnguleMathias opened the pull request:
#### What does this PR do?
* Users are able to receive descriptive signup/Registration validation errors

#### Description of Task to be completed
* Users should receive descriptive error messages upon signing up with invalid user parameters 
* Invalid user parameters include:
                  i.   Existing user email
                  ii.  Invalid email
                  iii. Password of length less than 8
                  iv. Password not being alphanumeric

#### How should this be manually tested?
* Run app by : `python3 migrate.py runserver` in terminal
* Use postman on end-route: `api/users/`
* Signup/Register user(s)

#### What are the relevant pivotal tracker stories?
* #161255353 - [Users should receive descriptive signup/Registration validation errors](https://www.pivotaltracker.com/story/show/161255353)

AnguleMathias committed:
Feature(Registration Validation Errors): Users should receive descriptive signup/Registration validation errors

- Update email validation
- Update password validation
- Added test for empty password validation

[Feature #161255353]

meshnesh reviewed (changes requested):
<img width="1142" alt="screen shot 2018-10-30 at 11 48 37" src="https://user-images.githubusercontent.com/15923109/47706243-cae43480-dc39-11e8-8e09-6c8c30bb4aa9.png">
Passwords should not accept repetitive letters.

AnguleMathias committed:
Feature(Registration Validation Errors): Users should receive descriptive signup/Registration validation errors

- Update password validation not to accept repeating patterns more than twice

[Feature #161255353]

maxwellgithinji reviewed (approved):
LGTM

Jwesh reviewed (approved):
good job
